tests/models/whisper/test_modeling_whisper.py

```diff
@@ -1850,6 +1850,33 @@ def test_tiny_token_timestamp_generation(self):
 
         self.assertTrue(torch.allclose(generate_outputs.token_timestamps.to("cpu"), EXPECTED_OUTPUT))
 
+    @slow
+    def test_tiny_token_timestamp_batch_generation(self):
+        set_seed(0)
+        processor = WhisperProcessor.from_pretrained("openai/whisper-tiny")
+        model = WhisperForConditionalGeneration.from_pretrained("openai/whisper-tiny")
+        model.to(torch_device)
+        model.generation_config.alignment_heads = [[2, 2], [3, 0], [3, 2], [3, 3], [3, 4], [3, 5]]
+
+        input_speech = self._load_datasamples(4)
+        input_features = processor.feature_extractor(raw_speech=input_speech, return_tensors="pt").input_features.to(
+            torch_device
+        )
+
+        generate_outputs = model.generate(
+            input_features,
+            max_length=448,
+            return_timestamps=True,
+            return_token_timestamps=True,
+            num_beams=3,
+            num_return_sequences=2,
+        )
+
+        self.assertEqual(generate_outputs.sequences.shape, generate_outputs.token_timestamps.shape)
+
+        # should return num_samples*num_return_sequences (4*2)
+        self.assertEqual(len(generate_outputs.sequences), 8)
```

Comment:
Not super fan of not being able to compare it to the ground-truth, like done [here](https://github.com/huggingface/transformers/blob/892dd91edde5ad8e7ba47ab35778b179b527a59f/tests/models/whisper/test_modeling_whisper.py#L1843-L1848). I've tried to compute it with the original whisper but it doesn't seem straightforward.

Any ideas on how to easily compute ground-truth here @sanchit-gandhi

Reply:
In the interest of time, what you can do is batched generation with bs=2, and then simply confirm that the outputs from the first batch match what we got with the un-batched integration test [mentioned above](https://github.com/huggingface/transformers/blob/892dd91edde5ad8e7ba47ab35778b179b527a59f/tests/models/whisper/test_modeling_whisper.py#L1843-L1848)

Reply:
The issue here is to test if the beam-search code snippet is working, not the batch generation ! So the problem can't be resolved with what you mention right ?